src/builder.py

```diff
@@ -111,21 +112,30 @@ def _initialize_image_builder(init_config: state.BuilderInitConfig) -> None:
             timeout=10 * 60,
             env=os.environ,
         )  # nosec: B603
+    except subprocess.CalledProcessError as exc:
+        logger.error(
+            "Failed to initialize builder, code: %s, out: %s, err: %s",
+            exc.returncode,
+            exc.stdout,
+            exc.stderr,
+        )
+        raise ImageBuilderInitializeError from exc
     except subprocess.SubprocessError as exc:
         raise ImageBuilderInitializeError from exc
 
 
-def install_clouds_yaml(cloud_config: dict) -> None:
+def install_clouds_yaml(cloud_config: state.OpenstackCloudsConfig) -> None:
     """Install clouds.yaml for Openstack used by the image builder.
 
     Args:
         cloud_config: The contents of clouds.yaml parsed as dict.
     """
+    cloud_config_dict = cloud_config.model_dump()
     if not OPENSTACK_CLOUDS_YAML_PATH.exists():
-        OPENSTACK_CLOUDS_YAML_PATH.write_text(yaml.safe_dump(cloud_config), encoding="utf-8")
+        OPENSTACK_CLOUDS_YAML_PATH.write_text(yaml.safe_dump(cloud_config_dict), encoding="utf-8")
         return
-    if yaml.safe_load(OPENSTACK_CLOUDS_YAML_PATH.read_text(encoding="utf-8")) != cloud_config:
-        OPENSTACK_CLOUDS_YAML_PATH.write_text(yaml.safe_dump(cloud_config), encoding="utf-8")
+    if yaml.safe_load(OPENSTACK_CLOUDS_YAML_PATH.read_text(encoding="utf-8")) != cloud_config_dict:
```

Comment:
Out of curiosity: could they differ due to differences in letter case (i.e., upper/lower case)?

Reply:
Yup! but the casing matters in this case (such as OpenStack password)